src/Core/Mews.Fiscalizations.Core/Xml/XmlSerializer.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.IO;
+using System.Xml;
+using System.Xml.Serialization;
+using FuncSharp;
+
+namespace Mews.Fiscalizations.Core.Xml
+{
+    public sealed class XmlSerializer
+    {
+        public static XmlElement Serialize<T>(T value, XmlSerializationParameters parameters = null)
+            where T : class
+        {
+            var namespaceSerializer = new XmlSerializerNamespaces();
+            var namespaces = parameters.ToOption().FlatMap(d => d.Namespaces).Flatten();
+
+            foreach (var ns in namespaces)
+            {
+                namespaceSerializer.Add(ns.Prefix, ns.Url);
+            }
+
+            var xmlDocument = new XmlDocument();
+            var navigator = xmlDocument.CreateNavigator();
+            using (var writer = navigator.AppendChild())
+            {
+                var xmlSerializer = new System.Xml.Serialization.XmlSerializer(typeof(T));
+                xmlSerializer.Serialize(writer, value, namespaceSerializer);
+                writer.Flush();
+                writer.Close();
+            }
+            return xmlDocument.DocumentElement;
+        }
+
+        public static T Deserialize<T>(XmlElement xmlElement)
+            where T : class
+        {
+            using (var reader = new StringReader(xmlElement.OuterXml))
```

Comment:
Just noticed this is working with the string anyway, so I guess there's no advantage constructing the XmlElement just to extract the string out of it again? Now I'd say the previous implementation was correct.

Or is there really some difference I don't see?

Reply:
I don't think there is any difference, both implementations should work.
i will just revert the commit where i changed this.

Reply:
👍 They would both work, but creating XmlElement and just using as a container seems like a waster.